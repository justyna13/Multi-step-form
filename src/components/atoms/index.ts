export { FormStep } from "@/components/atoms/FormStep/FormStep.tsx";
export { Button } from "@/components/atoms/Button/Button.tsx";
export { Input } from "@/components/atoms/Input/Input.tsx";
export { Header } from "@/components/atoms/Header/Header.tsx";
