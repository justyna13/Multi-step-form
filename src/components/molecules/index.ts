export { FormNavigation } from "@/components/molecules/FormNavigation/FormNavigation.tsx";
export { Footer } from "@/components/molecules/Footer/Footer.tsx";
export { FormInput } from "@/components/molecules/FormInput/FormInput.tsx";
export { FormSummary } from "@/components/molecules/FormSummary/FormSummary.tsx";
