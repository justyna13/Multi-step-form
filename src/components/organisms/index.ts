export { FormStepFirst } from '@/components/organisms/FormStepFirst/FormStepFirst.tsx';
export { FormStepSecond } from '@/components/organisms/FormStepSecond/FormStepSecond.tsx';
export { FormStepThird } from '@/components/organisms/FormStepThird/FormStepThird.tsx';
export { FormStepFourth } from '@/components/organisms/FormStepFourth/FormStepFourth.tsx';
export { FormConfirmation } from '@/components/organisms/FormConfirmation/FormConfirmation.tsx';
