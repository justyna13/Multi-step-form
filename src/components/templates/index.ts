export { HomePageTemplate } from '@/components/templates/HomePageTemplate/HomePageTemplate.tsx';
