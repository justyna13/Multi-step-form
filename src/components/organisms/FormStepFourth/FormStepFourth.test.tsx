import { describe, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { FormStepFourth } from "@/components/organisms";

describe("FormStepFourth", () => {

  it("should render fourthStep", () => {
    render(
      <FormStepFourth testid={'multiStepForm-step-fourth'} />
    );
    const formFourthStep = screen.getByTestId('multiStepForm-step-fourth');
    expect(formFourthStep).toBeInTheDocument();
  })
})
