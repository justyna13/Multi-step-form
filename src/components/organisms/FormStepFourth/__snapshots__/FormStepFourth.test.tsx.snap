// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormStepFourth > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="multiStepForm-step-fourth"
      >
        <div
          class="header"
          data-testuid="multiStepForm-step-fourth-ttl"
        >
          <h2>
            Finishing up
          </h2>
          <p>
            Double-check everything looks OK before confirming.
          </p>
        </div>
        <div
          class="form-summary"
        >
          <div
            class="form-summary__inner"
          >
            <div
              class="form-summary__plan"
            >
              <div>
                <p>
                   (Monthly)
                </p>
                <span
                  class="link"
                >
                  Change
                </span>
              </div>
              <span>
                +$
                0
                /mo
              </span>
            </div>
          </div>
          <div
            class="form-summary__total"
          >
            <p>
              Total (per month/year)
            </p>
            <span>
              +$12/mo
            </span>
          </div>
        </div>
        <div
          class="step-bottom"
        >
          <button
            class="main-btn undefined btn-prev"
            type="button"
          >
            Go Back
          </button>
          <button
            class="main-btn undefined btn-next"
            type="submit"
          >
            Confirm
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="multiStepForm-step-fourth"
    >
      <div
        class="header"
        data-testuid="multiStepForm-step-fourth-ttl"
      >
        <h2>
          Finishing up
        </h2>
        <p>
          Double-check everything looks OK before confirming.
        </p>
      </div>
      <div
        class="form-summary"
      >
        <div
          class="form-summary__inner"
        >
          <div
            class="form-summary__plan"
          >
            <div>
              <p>
                 (Monthly)
              </p>
              <span
                class="link"
              >
                Change
              </span>
            </div>
            <span>
              +$
              0
              /mo
            </span>
          </div>
        </div>
        <div
          class="form-summary__total"
        >
          <p>
            Total (per month/year)
          </p>
          <span>
            +$12/mo
          </span>
        </div>
      </div>
      <div
        class="step-bottom"
      >
        <button
          class="main-btn undefined btn-prev"
          type="button"
        >
          Go Back
        </button>
        <button
          class="main-btn undefined btn-next"
          type="submit"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
